spec/controllers/assignments_controller_spec.rb

```diff
@@ -3,60 +3,55 @@
 RSpec.describe AssignmentsController, type: :controller do
 
   # Create user for login
-  before(:each) do
+  Given {
     @user = FactoryGirl.create(:user)
     sign_in @user
-  end
+  }
 
   # Test create action
   describe "post #create" do
-    before do
-      @deliverer = FactoryGirl.create_list(:deliverer, 2)
-      @shift = FactoryGirl.create(:shift)
-    end
+    Given!(:deliverer){ FactoryGirl.create_list(:deliverer, 2) }
```

Comment:
We should favor `Given` as the value is only calculated on the first call, which helps avoid setting up unnecessary data for some context.
`Given!` is used when we need the setup but the variable is not directly called in the subsequent statements.
(This is just FYI. It's not necessary to fix here.)

Reply:
Okay I'll take note of that in the future